src/codegen/opt/build_cuda_on.cc

```diff
@@ -26,11 +28,45 @@ namespace codegen {
     }                                                                   \
   }
 
-std::string NVRTCCompile(const std::string& code) {
+std::string NVRTCCompile(const std::string& code, bool include_path = false) {
+  const int PARAM_NUM = 2;
+  char *compileParams[PARAM_NUM];
   nvrtcProgram prog;
+  cudaDeviceProp deviceProp;
+  std::string cc = "30";
+  cudaError_t e = cudaGetDeviceProperties(&deviceProp, 0);
+
+  if (e == cudaSuccess) {
+    cc = std::to_string(deviceProp.major) + std::to_string(deviceProp.minor);
+  } else {
+    LOG(WARNING) << "cannot detect compute capability from your device, "
+                 << "fall back to compute_30.";
+  }
+
+  if (include_path) {
+    std::string archOption = "-arch=compute_" + cc;
+    std::string includeOption = "--include-path=";
+    const char* cudaHomePath = std::getenv("CUDA_HOME");
+
+    if (cudaHomePath != nullptr) {
+      includeOption += cudaHomePath;
+      includeOption += "/include";
```

Comment:
Does this work on Windows? I'm not sure.

Reply:
Oh, I overlooked that. I'll address it. thanks